pkg/network/tracer/tracer_linux_test.go

```diff
@@ -992,23 +997,30 @@ func (s *TracerSuite) TestDNATIntraHostIntegration() {
 	require.NoError(t, err, "error connecting to client")
```

Comment:
I think you want to change this to assert and move the net.Dial into the EventuallyWithT. It's not guaranteed that the server is accepting connections by the time server.Run() exist because `accept()` is called in another goroutine. 

Alternatively, we might want to have `server.Run()` have stronger guarantees about the server accepting connections.